R/modify_legend.R

```diff
@@ -0,0 +1,191 @@
+
+#' add function call to the overall legend
+#' 
+#' @param object
+#' @param fun.name
+#' @param legend.name
+#' @param \dots
+#' @examples 
+#' gs <- gsplot() %>% 
+#'          points(x=1:5, y=1:5, legend.name = 'points 1') %>% 
+#'          legend()
+#' gs
+#' 
+#' gs <- gsplot() %>% 
+#'          points(x=1:2, y=1:2, col = c("red", "blue"), 
+#'                 legend.name = c('points 1', 'points 2')) %>% 
+#'          legend()
+#' gs
+#' 
+#' @keywords internal
+add_to_legend <- function(object, fun.name, legend.name, ...){
```

Comment:
@lindsaycarr can you help me figure out how to plug this in? 

For example, should a function like `points` call both `modify_legend` and `add_to_lengend`?